infrastructure/config/config_yaml.py

```diff
@@ -80,19 +84,11 @@ def __init__(self, filename: str) -> None:
         except Exception as err:
             raise Exception("Configuration error") from err
 
-    def _parse_logging_level(self, config) -> int:
-        try:
-            if config["logging_level"] == "FATAL" or config["logging_level"] == "CRITICAL":
-                return logging.CRITICAL
-            if config["logging_level"] == "ERROR":
-                return logging.ERROR
-            if config["logging_level"] == "WARN" or config["logging_level"] == "WARNING":
-                return logging.WARNING
-            if config["logging_level"] == "INFO":
-                return logging.INFO
-            if config["logging_level"] == "DEBUG":
-                return logging.DEBUG
-            else:
-                raise ValueError(f"{config['logging_level']} is not defined")
-        except KeyError:
-            return logging.INFO
+    def _parse_logging_level(self, level_str: str) -> int:
+        return {
+            "CRITICAL": logging.CRITICAL,
+            "ERROR": logging.ERROR,
+            "WARNING": logging.WARNING,
+            "INFO": logging.INFO,
+            "DEBUG": logging.DEBUG,
+        }.get(level_str, logging.DEBUG)
```

Comment:
That is a cunning idea, however:
- we need to notify the operator that they've mistyped the name of the log level in the config, as opposed to silently using a different level. With your current solution, if an operator puts `INOF` instead of `INFO` in the config, they will end-up logging at DEBUG level (and thinking how oddly verbose the INFO level is :))
- the default log level should not be `DEBUG`
- you've unsupported log levels `WARN` and `FATAL`, so anybody who used one of them will end up with `DEBUG`.
This code should look like this:
```
def _parse_logging_level(self, level_str: str) -> int:
    try:
        return {
            "CRITICAL": logging.CRITICAL,
            "FATAL": logging.CRITICAL,
            "ERROR": logging.ERROR,
            "WARNING": logging.WARNING,
            "WARN": logging.WARNING,
            "INFO": logging.INFO,
            "DEBUG": logging.DEBUG,
        }[level_str.upper()]
    except KeyError:
        raise ValueError(f"unknown loggging level '{level_str}'")
```
(I've also made the `logging_level` values case insensitive)

Reply:
I would stick to narrow set of possible values of [CRITICAL, ERROR, WARNING, INFO, DEBUG] but OK.